packages/jellyfish-api-core/__tests__/category/loan/withdrawFromVault.test.ts

```diff
@@ -0,0 +1,420 @@
+
+      { // after first withdrawal
+        const accountBalances = await tGroup.get(0).rpc.account.getAccount(destinationAddress)
+        expect(accountBalances.length).toStrictEqual(1)
+        expect(accountBalances[0]).toStrictEqual('9.87600000@DFI')
+
+        const { collateralAmounts } = await anotherMn.rpc.loan.getVault(vaultId1) as VaultActive
+        expect(collateralAmounts?.length).toStrictEqual(2)
+
+        {
+          const dfiCol = collateralAmounts?.find(c => c.includes('DFI'))
+          expect(dfiCol).toBeDefined()
+          const [amount, tokenSymbol] = (dfiCol as string).split('@')
+          expect(tokenSymbol).toStrictEqual('DFI')
+          expect(amount).toStrictEqual('9990.12400000')
+        }
+        { // unchanged
+          const btcCol = collateralAmounts?.find(c => c.includes('BTC'))
+          expect(btcCol).toBeDefined()
+          const [amount, tokenSymbol] = (btcCol as string).split('@')
+          expect(tokenSymbol).toStrictEqual('BTC')
+          expect(amount).toStrictEqual('1.00000000')
+        }
+      }
+
+      await tGroup.get(0).rpc.loan.withdrawFromVault({
+        vaultId: vaultId1,
+        to: destinationAddress,
+        amount: '0.123@BTC'
+      })
+      await tGroup.get(0).generate(1)
+      await tGroup.waitForSync()
+
+      { // after second withdrawal
+        const accountBalances = await anotherMn.rpc.account.getAccount(destinationAddress)
+        expect(accountBalances.length).toStrictEqual(2)
+        expect(accountBalances[0]).toStrictEqual('9.87600000@DFI')
+        expect(accountBalances[1]).toStrictEqual('0.12300000@BTC')
+
+        const { collateralAmounts } = await anotherMn.rpc.loan.getVault(vaultId1) as VaultActive
+        expect(collateralAmounts?.length).toStrictEqual(2)
+
+        {
+          const dfiCol = collateralAmounts?.find(c => c.includes('DFI'))
+          expect(dfiCol).toBeDefined()
+          const [amount, tokenSymbol] = (dfiCol as string).split('@')
+          expect(tokenSymbol).toStrictEqual('DFI')
+          expect(amount).toStrictEqual('9990.12400000')
+        }
+        { // unchanged
+          const btcCol = collateralAmounts?.find(c => c.includes('BTC'))
+          expect(btcCol).toBeDefined()
+          const [amount, tokenSymbol] = (btcCol as string).split('@')
+          expect(tokenSymbol).toStrictEqual('BTC')
+          expect(amount).toStrictEqual('0.87700000')
+        }
+      }
+    })
+
+    it('should withdrawFromVault using specific utxo', async () => {
+      const utxo = await tGroup.get(0).container.fundAddress(vaultOwner, 10)
+      const txid = await tGroup.get(0).rpc.loan.withdrawFromVault({
+        vaultId: vaultId1,
+        to: collateralAddress,
+        amount: '2.34567@DFI'
+      }, [utxo])
+
+      const rawtx = await tGroup.get(0).container.call('getrawtransaction', [txid, true])
+      expect(rawtx.vin[0].txid).toStrictEqual(utxo.txid)
+      expect(rawtx.vin[0].vout).toStrictEqual(utxo.vout)
```

Comment:
maybe add a line to check `collateralAddress` received the withdrawn amount for completeness.